catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/UpdateOperations.java

```diff
@@ -733,4 +736,37 @@ private UpdateStorageRequest applyAttributeOverrides(UpdateStorageRequest update
                 metacardMap);
         return updateStorageRequest;
     }
+
+    private UpdateRequest populateResourceUris(UpdateRequest updateRequest, QueryResponse query) {
+        Set<Metacard> updateRequestMetacards = updateRequest.getUpdates()
+                .stream()
+                .map(Map.Entry::getValue)
+                .filter(metacard -> metacard.getResourceURI() == null)
+                .collect(Collectors.toSet());
+
+        Map<String, Metacard> queryResultMetacards = query.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .filter(Objects::nonNull)
+                .filter(metacard -> metacard.getResourceURI() != null)
+                .collect(Collectors.toMap(Metacard::getId, Function.identity()));
+
+        for (Metacard updateMetacard : updateRequestMetacards) {
+            Metacard queriedMetacard = queryResultMetacards.get(updateMetacard.getId());
+            if (queriedMetacard != null && queriedMetacard.getId()
+                    .equals(updateMetacard.getId())) {
+                LOGGER.info(
+                        "Found URI \"{}\" missing from updated Metacard with ID \"{}\". Setting Metacard field \"{}\" with the value.",
```

Comment:
@brendan-hofmann It is possible to remove attributes as needed so I am not sure how you would tell the difference between dropping and removing currently.  Content resource URIs are a special case that we do not allow either way.  I do see things like this making it more difficult in the future to support partial updates but I don't think we need to worry about that use case for this PR.